packages/node_modules/@webex/plugin-meetings/src/meetings/index.js

```diff
@@ -534,61 +567,32 @@ export default class Meetings extends WebexPlugin {
      * @memberof Meetings
      */
     destroy(meeting, reason) {
-      if (this.config.autoUploadLogs) {
-        LoggerProxy.logger.info('Meetings:index#destroy --> Meeting destroyed, uploading logs');
-        const callStart = meeting.locusInfo && meeting.locusInfo.fullState ? meeting.locusInfo.fullState.lastActive : undefined;
+      const deleteMeeting = () => {
+        this.meetingCollection.delete(meeting.id);
+        Trigger.trigger(
+          this,
+          {
+            file: 'meetings',
+            function: 'destroy'
+          },
+          EVENT_TRIGGERS.MEETING_REMOVED,
+          {
+            meetingId: meeting.id,
+            reason
+          }
+        );
+      };
 
+      if (this.config.autoUploadLogs) {
         this.uploadLogs({
-          callStart,
+          callStart: meeting.locusInfo && meeting.locusInfo.fullState ? meeting.locusInfo.fullState.lastActive : undefined,
           correlationId: meeting.correlationId,
           feedbackId: meeting.correlationId,
-          locusId: meeting.locusId
-        })
-          .then((uploadResult) => {
-            LoggerProxy.logger.info('Meetings:index#destroy --> Upload logs for meeting completed.', uploadResult);
-            Trigger.trigger(
-              this,
-              {
-                file: 'meetings',
-                function: 'destroy'
-              },
-              EVENT_TRIGGERS.MEETING_LOG_UPLOAD_SUCCESS,
-              {
-                meetingId: meeting.id,
-                details: uploadResult
-              }
-            );
-          })
-          .catch((uploadError) => {
-            LoggerProxy.logger.error('Meetings:index#destroy --> Unable to upload logs for meeting', uploadError);
-            Trigger.trigger(
-              this,
-              {
-                file: 'meetings',
-                function: 'destroy'
-              },
-              EVENT_TRIGGERS.MEETING_LOG_UPLOAD_FAILURE,
-              {
-                meetingId: meeting.id,
-                reason: uploadError
-              }
-            );
-          });
+          locusId: meeting.locusId,
+          meetingId: meeting.id
+        }).finally(() => deleteMeeting());
```

Comment:
We shouldn't need this `finally` line. We are passing the info needed to the upload function and we don't need to wait for the update before destroying the meeting. Plus with `deleteMeeting()` being called on 595, it would be called immediately anyway.